File: day-1/src/Day1Part2.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

public class Day1Part2 {
    public static void main(String[] args) {
        int sum = 0;
        int k;
        ArrayList<Integer> list = new ArrayList<Integer>();
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();

        File file = new File("src/input");
        try {
            Scanner sc = new Scanner(file);
            while (sc.hasNextLine()) {
                String line = sc.nextLine();
                try {
                    // Split and parse the line
                    int left = Integer.parseInt(line.split(" ")[0]);
                    int right = Integer.parseInt(line.split(" ")[3]);
                    System.out.println(left + " " + right);

                    // Add to list and map
                    list.add(left);
                    map.put(right, map.getOrDefault(right, 0) + 1);
                } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
                    System.err.println("Skipping invalid line: " + line);
                }
            }
            sc.close();
        } catch (FileNotFoundException e) {
            System.err.println("File not found: " + file.getAbsolutePath());
            return;
        }



        for (int n: list) {
            int count = map.getOrDefault(n, 0);
            sum += n * count;
            System.out.println("list(i): " + count + ". Times this number has been seen: " + list.get(count) + ". Sum: " + sum);
        }
        System.out.println(sum);
    }
}

File: Python/day-1/day_1_part_2.py
def main():
    sum = 0
    leftLst = []
    dct = {}

    # Parse file into two lists.
    with open("input") as f:
        for line in f.readlines():
            splt = line.split()
            leftLst.append(int(splt[0]))
            rightNum = int(splt[1])
            # Increment dict value or set new value to 1
            dct[rightNum] = dct.get(rightNum, 0) + 1

    # Access key value
    for num in leftLst:
        sum += num * dct.get(num, 0)
        print(f"leftLst[i] = {num}, times this number has been seen = {dct.get(num)}, sum = {sum}")

    print(f"sum = {sum}")


if __name__ == '__main__':
    main()
